src/aws-cpp-sdk-core/source/client/AWSXmlClient.cpp

```diff
@@ -186,9 +186,19 @@ AWSError<CoreErrors> AWSXMLClient::BuildAWSError(const std::shared_ptr<Http::Htt
     {
         auto responseCode = httpResponse->GetResponseCode();
         auto errorCode = AWSClient::GuessBodylessErrorType(responseCode);
-
         Aws::StringStream ss;
-        ss << "No response body.";
+        
+        if(!httpResponse->GetResponseBody())
+        {
+            ss << "No response body.";
+        }
+        else
+        {
+            Aws::String xmlString((Aws::IStreamBufIterator(httpResponse->GetResponseBody())), Aws::IStreamBufIterator());
+
+            ss <<xmlString;
+            ss << " current pointer location in stream = "<<httpResponse->GetResponseBody().tellp();
```

Comment:
instead of returning the pointer location lets just stilll return `No response body.` and log out a debug statement. The message consumed as a error message in a response could be somewhat confusing.